### DISCUSSION TITLE
Pool creation timeout

### DISCUSSION THREAD

**Discussion by PaulWasTaken** [oracle/python-oracledb#84] [Q&A] [answered]
Hello, could you point me out about pool initializing.
I wonder how pool is created, but I am not that good in Cython.
I noticed that pool creation time depends on `TRANSPORT_CONNECT_TIMEOUT`, so I expected an log messages when this param is lower than expected. There is https://github.com/oracle/python-oracledb/blob/main/src/oracledb/impl/thick/pool.pyx#L147 here, but it is never thrown
The approach is "We want to acquire some connections (`min` param) beforehand (in create_pool), but if we fail, we will establish connections on requests"
Am I right?

**Comment by anthony-tuininga** [comment, marked as answer]
First, there are some differences between thick mode and thin mode in how pools are initialized. In thin mode, the connections are established in a background thread and failure to create these connections will be reported when the first call to `pool.acquire()` is made. Thick mode, however, currently establishes `min` connections and if any of those fails, the call to `oracledb.create_pool()` itself will fail. This may change at some point to use a background thread similarly to thin mode.

The value of the parameter `TRANSPORT_CONNECT_TIMEOUT` indicates how long to wait for the initial socket connection to be established with the listener. I'm not sure what log messages you were expecting. Can you clarify?

**Comment by PaulWasTaken** [comment]
> I'm not sure what log messages you were expecting.

For some reason I thought it would be reported as soon as possible.

I was confused because the script run twice long than expected. I thought it is because of `TRANSPORT_CONNECT_TIMEOUT` - the connection was tried to establish two times

UPD: I figured it out - if `pool.acquire()` is called before timeout first we will wait for `TRANSPORT_CONNECT_TIMEOUT`
